torch/package/find_file_dependencies.py

```diff
@@ -57,47 +57,46 @@ def _grab_node_str(self, node):
     def visit_Call(self, node):
         # __import__ calls aren't routed to the visit_Import/From nodes
         if hasattr(node.func, "id") and node.func.id == "__import__":
-            if type(node.args[0]) not in [ast.Constant, ast.Str]:
-                # We don't want to parse dynamic uses of __import__
+            try:
```

Comment:
can we narrow the check? we really only want to catch when the imported module isn't constant, any other exceptions should get raised properly. It seems like the existing check only checks the first argument for constantness—maybe the narrowest check would be to check all arguments for constantness and skip if any of them aren't.

Reply:
That might be too strict of a check -- The global/local variables aren't constants normally ex: `module = __import__(module_name, globals(), locals(), [''])`

Reply:
alright, it's probably not so bad anyway since the try-catch only covers `__import__` calls